### PR TITLE
feat(cli): scaffold `totem proposal new` and `totem adr new` (#1288)

### LINKED ISSUE #1288: Add totem proposal and totem adr entity commands for scaffolded authoring
## Summary

Add `totem proposal` and `totem adr` entity commands for scaffolded authoring of strategy proposals and architecture decision records. Both commands solve the same class of friction: structured governance-artifact authoring that currently requires tribal knowledge about filename conventions, template shapes, dashboard regeneration, and PR workflow.

## The friction (experienced in the wild during PR #1287 + totem-strategy#77)

Writing a one-line cross-reference fix to `proposals/active/218-governance-fine-tuning.md` during end-of-day signoff required me to:

1. **Know the filename format**: `proposals/active/NNN-kebab-title.md` where NNN is auto-incremented but the increment authority is just "inspect the directory"
2. **Know the template shape**: Status / Author / Date / Milestone sections at top, Motivation / Problem Statement / Proposed Solution / Consequences / Decision Needed sections below
3. **Know to run `pnpm run docs:inject` after any proposal add/delete** so the auto-generated PROPOSAL_INBOX section in README.md refreshes
4. **Know the feature-branch-then-PR workflow** for the strategy repo (main protection with `[totem-strategy] Direct commits to 'main' are blocked.` hook)
5. **Discover via push-failure that the dashboard was stale** (the pre-push hook told me, but I should have known to regen first)

All five of those are tribal knowledge a new contributor would have to reverse-engineer. The same applies to ADRs — the `ADR_TABLE` section of README.md has the same staleness failure mode, and the file format `adr/adr-NNN-kebab-title.md` has its own numbering/template conventions.

## Proposed commands

```bash
# Scaffold a new proposal with auto-incremented number and template
totem proposal new "LLM context caching for verification layer"
  # → .strategy/proposals/active/219-llm-context-caching-for-verification-layer.md
  # → pre-filled with Status: Draft, Author: (from git config or totem.config.ts),
  #   Date: today, Milestone: (prompt or config default)
  # → runs docs:inject automatically to refresh the PROPOSAL_INBOX dashboard
  # → stages + commits the new file and the dashboard update
  # → optionally: opens a feature branch + PR

# List all proposals with status
totem proposal list
  # | 212 | Model Routing Matrix             | Draft    |
  # | 213 | Context Tuning Initiative        | Draft    |
  # | 214 | Lite Binary Size Gate            | Accepted |
  # ...
  # Useful for "what's the next available number?" and triage

# Transition a proposal's status
totem proposal status 214 "Accepted (2026-04-07)"
  # Updates the status line in the proposal file
  # Re-runs docs:inject to update the dashboard

# ADRs get identical treatment
totem adr new "Totem Pack Ecosystem"
totem adr list
totem adr status 085 "Accepted"
```

## Why this fits the totem philosophy

1. **It's the same problem as `totem lesson add`** (which already exists). Both are structured authoring workflows for a governance artifact. Having lesson tooling without proposal/adr tooling is a gap.

2. **It closes the "dashboard staleness" hook loop structurally.** Right now, the pre-push hook in totem-strategy catches stale dashboards but forces a manual `docs:inject` recovery. If `totem proposal new` runs `docs:inject` as part of its commit, the dashboard never goes stale in the first place. Same architectural pattern as the wind tunnel SHA recovery ticket (#1285) — fix the workflow to not produce the stale state rather than catch it later.

3. **It unlocks LLM-assisted proposal authoring.** Gemini drafted Proposals 216, 217, and 218 today by editing files directly. If there were a `totem proposal new --from-outline <markdown>` flag that accepted a rough outline and filled in the template, Gemini (or future Claude) could produce proposals via a single command instead of a multi-file dance.

## Scope for initial implementation (1.14.0)

**Minimum viable:**
- `totem proposal new <title>` — scaffold + docs:inject + commit (no PR opening)
- `totem adr new <title>` — same for ADRs
- Templates live in `.strategy/templates/proposal.md` and `.strategy/templates/adr.md` (or inherit defaults if missing)
- Auto-increment number by scanning the target directory

**Deferred to 1.15.0+:**
- `list` subcommands (informational, lower priority)
- `status` subcommands (transition workflows)
- `--from-outline` flag for LLM-assisted drafting
- PR-opening integration
- Cross-repo discovery (works across linked totems)

## Cross-references to related tickets

- **#1274** (`handoff` output destination not configurable) — filed by playground Claude during PR #33. Same class of friction: hardcoded paths for governance artifacts that should be configurable/scaffolded. The `totem proposal` design should inform the `totem handoff` fix.
- **#1284** (MCP add_lesson double-heading bug) — filed earlier today. Authoring-DX friction in the lesson space. The fix direction for #1284 (explicit heading handling) should inform the proposal/adr scaffolding contract.
- **#1285** (wind tunnel SHA hook ordering) — related structural-fix-vs-recovery pattern.

## Open design questions

- Should the commands operate on the strategy submodule (when present in the totem repo workflow) or directly on the strategy repo (standalone)? Both cases should work.
- How does the next-number computation handle gaps (e.g., Proposal 216 was reverted — does 216 become reusable, or does the counter skip to 219)?
- Should the commands require a `.strategy/templates/` directory, or fall back to hardcoded defaults?
- Should `list` query the dashboard markdown tables, or re-scan the directory each invocation?

These are implementation details, not blockers — the core scaffolding workflow is clearly valuable regardless.

## Discovered during

End-of-day review of the two new proposals (217 + 218) on 2026-04-07. I hit the exact friction writing the one-line cross-reference fix for Proposal 218 (merged as `mmnto-ai/totem-strategy#77`), which required me to manually understand the file format, template, dashboard regen step, and PR workflow — none of which should be required for a one-line fix.

Per the end-of-day signoff discussion: Gemini's directive was **"Claude's idea for scaffold commands is brilliant. It perfectly solves the dashboard staleness issue by automating the boilerplate. But it is a workflow chore, not an architectural blocker. File it as a feature ticket in the 1.14.0 DX bucket so it doesn't get lost, but do not let it derail the Context Mesh epic."**

## Acceptance criteria

- [ ] `totem proposal new <title>` creates a valid proposal file with auto-incremented number
- [ ] `totem adr new <title>` creates a valid ADR file with auto-incremented number
- [ ] Dashboard tables in README.md are automatically refreshed after either command
- [ ] New file is staged in git (but not committed without user confirmation)
- [ ] Command works from both the strategy repo and the submodule path (when run from within the totem repo)
- [ ] Templates are configurable via `.strategy/templates/` directory with fallback defaults
- [ ] At least 3 tests: happy path for each command + one "gap number" edge case

### PR DESCRIPTION
**Pull request by satur8d**
## Summary

- `totem proposal new <title>` and `totem adr new <title>` scaffold NNN-prefixed artifacts under the strategy repo layout, fill a template matching ADR-091's heading convention (`# ADR NNN: Title` with space separator), refresh the dashboard via `pnpm run docs:inject`, and stage the two mutated paths.
- Auto-increment respects numbering gaps (`001-a.md` + `003-b.md` yields `004-c.md`); slug sanitization strips non-alphanumerics; empty-slug titles throw loud before any disk write; exact-filename collisions throw without overwriting.
- Supports both the submodule case (`<totem>/.strategy/`) and the standalone case (cwd is the strategy repo itself).
- Warn-and-continue on `docs:inject` and `git add` failures so a scaffolded artifact never gets stranded without feedback.
- Default template carries `Status: Draft` + `Date: YYYY-MM-DD` frontmatter, matching the ADR-091 convention surfaced during this session's manual ADR-092 promotion cycle.

## Scope

MVP only per the ticket's "Minimum viable" section. Deferred: `list` / `status` / `promote` subcommands, `--from-outline` LLM-assisted drafting, PR-opening integration, `--force`, `--no-stage`, cross-repo discovery flags. The `promote` state transition (Draft → Accepted + file move) is the obvious next lap once the MVP settles.

## Design doc

`.totem/specs/1288.md` carries the preflight spec plus the Implementation Design section (scope, data model deltas, state lifecycle, failure modes table, invariants locked via tests, open questions). The four open questions were resolved per user direction before coding started.

## Follow-up ticket

**#1614** filed for refining the `lesson-fail-open-catch-ban` ast-grep rule so it recognizes terminal handlers like `handleError` (declared `: never`) without needing a per-site `// totem-context:` annotation. Every new CLI command added to `index.ts` hits this false positive today. Workaround is in place (trailing comment on the try body's last statement) but the rule itself should learn the pattern.

## Verification

- `totem lint`: PASS (0 errors, 223 warnings; all either test-fixture patterns or rule false-positives on production code)
- `totem review`: PASS (0 findings after one revision round that caught a real `$&` back-reference bug in the template renderer, fixed + regression test)
- `totem verify-manifest`: PASS (440 rules, hashes match)
- Test suite: full monorepo green, including `@totem/pack-agent-security` repo-sweep with the new governance.ts spawn sites allowlisted

## Test plan

- [x] Unit: `governance.test.ts` — 28 tests covering path resolution (submodule + standalone), auto-increment with gap, filename sanitization + collision + empty-slug, template variable substitution including `$&` back-reference safety, post-scaffold hook argv, docs:inject/git-add failure resilience, and orchestrator end-to-end
- [x] Unit: `proposal.test.ts` + `adr.test.ts` — 4 tests covering CLI adapter happy path in both submodule and standalone contexts plus gap numbering and ADR-091 heading form
- [x] Pack sweep: `packages/pack-agent-security/test/repo-sweep.test.ts` allowlist entry for `governance.ts` spawn sites (expected count 2)
- [ ] Manual: run `totem proposal new "Test"` against the strategy submodule and confirm the generated file matches the ADR-091 heading convention byte-for-byte
- [ ] Manual: run `totem adr new "Test"` and confirm the same

Closes #1288.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: packages/cli/src/utils/governance.ts [hunk 1]
```diff
@@ -0,0 +1,426 @@
...
+    template = type === 'proposal' ? DEFAULT_PROPOSAL_TEMPLATE : DEFAULT_ADR_TEMPLATE;
+  }
+
+  // Replacer-function form avoids the `$&` / `$1` back-reference trap in
+  // `String.prototype.replace`'s string-replacement mode. A title like
+  // `Fix $foo bug` would otherwise mis-render. See PR #1429 review cycle.
+  return template
+    .replace(/\{\{TITLE\}\}/g, () => title)
+    .replace(/\{\{DATE\}\}/g, () => date)
+    .replace(/\{\{ID\}\}/g, () => id);
```
[the remark this answers is not in the document]
Comment (satur8d) [reply packages/cli/src/utils/governance.ts]: @gemini-code-assist thanks for the catch. Fixed in `eece2e2c` by collapsing `renderArtifactTemplate` to a single-pass regex with a keyed replacer function, matching your snippet almost verbatim. Regression test added: `title = '{{DATE}}'` now renders literally while a distinct `{{DATE}}` in the template body still resolves. Preserves the replacer-function safety against `$&` / `$1` interpretation from the previous fix.

File: packages/cli/src/utils/governance.ts [hunk 2]
```diff
@@ -0,0 +1,429 @@
...
+ * layout can be detected.
+ */
+export function resolveGovernancePaths(cwd: string, type: GovernanceType): GovernancePaths {
+  const gitRoot = resolveGitRoot(cwd);
+  if (gitRoot === null) {
+    throw new TotemError(
+      'CONFIG_MISSING',
+      `Not inside a git repository: ${cwd}`,
+      'Run this command from inside a Totem or Totem-strategy repository checkout.',
+    );
```
[the remark this answers is not in the document]
Comment (satur8d) [reply packages/cli/src/utils/governance.ts:66]: @gemini-code-assist declining this one. Rule 32 in `.gemini/styleguide.md` references `throw new Error('[Totem Error] ...')` (plain `Error`). For `TotemError`, the centralized `handleError` in `packages/cli/src/index.ts` already handles the prefix — it checks `err.message.startsWith('[Totem Error]')` and prepends only when missing, so manual prefixes would either duplicate (if added naively) or be silently deduped (current behavior).

Every existing `new TotemError(...)` call in the codebase passes the plain message without manual prefix (see `packages/cli/src/adapters/gh-utils.ts:57`, `packages/cli/src/adapters/multi-repo-adapter.ts:44,54`, `packages/cli/src/commands/check.ts:34`, `packages/cli/src/commands/compile.ts:406,454,493,514,537`). Keeping consistency with that convention.

This is a known pattern on your side: the same finding came up on PR #1544 and PR #1553 and was conceded both times. No code change.

File: packages/cli/src/utils/governance.ts [hunk 3]
```diff
@@ -0,0 +1,429 @@
...
+
+  let staged = false;
+  try {
+    exec('git', ['add', newFilePath, dashboardFile], rootDir);
+    staged = true; // totem-context: intentional warn-and-continue per spec 1288 — git add failure must not strand the scaffolded artifact on disk.
+  } catch (err) {
+    const msg = err instanceof Error ? err.message : String(err);
+    // eslint-disable-next-line no-console -- intentional stderr warning per spec
+    console.warn(
+      `[Totem Warning] git add failed (${msg}). File created but not staged; run 'git add' manually.`,
```
[the remark this answers is not in the document]
Comment (satur8d) [reply packages/cli/src/utils/governance.ts]: @coderabbitai fixed in `a0331d7b`. Switched both warn-and-continue paths from `console.warn` + `eslint-disable-next-line no-console` to `log.warn('Totem', ...)` from `packages/cli/src/ui.ts`. That helper already writes to stderr with the canonical `[Totem]` prefix, so no more unticketed suppressions. Existing tests updated to spy on `console.error` (where `log.warn` routes) instead of `console.warn`.

Declining the "thread a warning sink from the command layer" half of the suggestion — that's a larger refactor across every CLI command that emits warnings and out of scope for this MVP. If you want it tracked, I can file a follow-up ticket.

File: packages/cli/src/utils/governance.ts [hunk 4]
```diff
@@ -0,0 +1,429 @@
...
+    id,
+    title: options.title,
+    templatePath: paths.templatePath,
+    date,
+  });
+
+  // Ensure parent dir exists (it should, from resolveGovernancePaths, but
+  // cheap to defend against a manually-pruned target).
+  fs.mkdirSync(paths.targetDir, { recursive: true });
+  fs.writeFileSync(filePath, rendered, 'utf-8');
```
[the remark this answers is not in the document]
Comment (satur8d) [reply packages/cli/src/utils/governance.ts]: @coderabbitai fixed in `a0331d7b`. New `sanitizeArtifactTitle` helper strips ASCII C0 controls (`0x00-0x1F`) and DEL (`0x7F`), then collapses runs of whitespace to a single space and trims. The orchestrator (`scaffoldGovernanceArtifact`) sanitizes at the entry point so both the filename slug and the template body render against the cleaned form.

Kept the `#` character through on purpose: a single-line h1 heading that contains `#` chars stays a valid single heading in markdown — the injection vector is the newline that breaks the line into separate blocks, not the `#` itself.

Regression tests added:
- 4 unit tests for `sanitizeArtifactTitle` covering newlines / tabs / CR / BEL / DEL / whitespace collapse / trimming / preservation of `#` and `$` and `{{ }}`.
- 1 orchestrator integration test that feeds a title containing `\n## Injected Fake Heading\n- Not a real list` and asserts the scaffolded file's heading is a single line with no newline-bounded blocks injected.

File: packages/cli/src/utils/governance.ts [hunk 5]
```diff
@@ -0,0 +1,459 @@
...
+    title: cleanTitle,
+    templatePath: paths.templatePath,
+    date,
+  });
+
+  // Ensure parent dir exists (it should, from resolveGovernancePaths, but
+  // cheap to defend against a manually-pruned target).
+  fs.mkdirSync(paths.targetDir, { recursive: true });
+  fs.writeFileSync(filePath, rendered, 'utf-8');
+
```
[the remark this answers is not in the document]
Comment (satur8d) [reply packages/cli/src/utils/governance.ts:461]: @coderabbitai fixed in `6caf7df3`. Added `flag: 'wx'` to the `writeFileSync` call + EEXIST catch-and-rethrow. The pre-check stays for a cleaner error on the common (non-racing) path; the wx flag is the authoritative guard for the race window.

Also filed **#1616** as the follow-up for the warning-sink refactor you asked about on the eslint-disable thread. Tracks the cross-cutting sweep so the `log.warn` change in this PR stays narrow and the broader refactor doesn't get lost.

No TOCTOU-specific integration test — Node's `fs` exports are non-configurable through `vi.spyOn`, and `vi.mock` for the whole module would break the other tests in this file. The wx behavior is a Node platform guarantee and the code diff is small enough for direct review.

File: packages/cli/src/utils/governance.ts [hunk 6]
```diff
@@ -0,0 +1,459 @@
...
+ *
+ * Throws `TotemError` when cwd is not inside a git repo, or when neither
+ * layout can be detected.
+ */
+export function resolveGovernancePaths(cwd: string, type: GovernanceType): GovernancePaths {
+  const gitRoot = resolveGitRoot(cwd);
+  if (gitRoot === null) {
+    throw new TotemError(
+      'CONFIG_MISSING',
+      `Not inside a git repository: ${cwd}`,
```
[the remark this answers is not in the document]
Comment (satur8d) [reply packages/cli/src/utils/governance.ts:64]: @gemini-code-assist declining all four `[Totem Error]` prefix findings on this pass (lines 64, 137, 162, 427) for the same reason as your prior concession on this PR.

**Direct self-contradiction:** [You conceded 15 minutes ago](https://github.com/mmnto-ai/totem/pull/1615#discussion_r3124575008) on line 66 — the same class of finding, the same file, the same `TotemError` throws: _"I acknowledge that the TotemError messages should remain plain to avoid duplication or deduplication issues given the current implementation of handleError."_ That reasoning applies equally to all other `new TotemError(...)` call sites in the file, which is why they all use the same plain-message form.

**On the `TotemConfigError` switch:** that class (exported from `packages/core/src/errors.ts`) is used specifically for schema validation in `packages/core/src/config-schema.ts` and the embedders — not for CLI-layer runtime validation. No `packages/cli/src/**` file uses `TotemConfigError`; every sibling validator in the CLI layer (`packages/cli/src/adapters/gh-utils.ts`, `packages/cli/src/commands/check.ts`, `packages/cli/src/commands/compile.ts`) uses plain `TotemError`. Keeping consistency with that layer.

No code change on this round.

File: packages/cli/src/utils/governance.ts [hunk 7]
```diff
@@ -0,0 +1,459 @@
...
+  let dashboardRefreshed = false;
+  try {
+    exec('pnpm', ['run', 'docs:inject'], rootDir);
+    dashboardRefreshed = true; // totem-context: intentional warn-and-continue per spec 1288 — dashboard refresh failure must not strand the scaffolded artifact on disk.
+  } catch (err) {
+    const msg = err instanceof Error ? err.message : String(err);
+    log.warn(
+      'Totem',
+      `docs:inject did not run cleanly (${msg}). Dashboard not refreshed; run 'pnpm run docs:inject' manually.`,
+    );
```
[the remark this answers is not in the document]
Comment (satur8d) [reply packages/cli/src/utils/governance.ts:338]: @gemini-code-assist declining this one. This directly contradicts [CodeRabbit's concession on the previous pass](https://github.com/mmnto-ai/totem/pull/1615#discussion_r3124576334), where `log.warn('Totem', ...)` from `packages/cli/src/ui.ts` was confirmed as the canonical stderr sink that resolves the no-console suppression concern.

The "utility modules should use standard console logging" framing would require reverting the fix CR just accepted. The cross-cutting sweep (whether utility modules should thread a shared warning sink vs. call `log.warn` directly vs. use `console.warn`) is tracked as **#1616** for a dedicated PR. This MVP stays narrow.

No code change on this round.
